x-pack/legacy/plugins/integrations_manager/server/packages/install.ts

```diff
@@ -8,35 +8,40 @@ import { SavedObject, SavedObjectsClientContract } from 'src/core/server/';
 import { SAVED_OBJECT_TYPE } from '../../common/constants';
 import { AssetReference, AssetType, InstallationAttributes } from '../../common/types';
 import * as Registry from '../registry';
-import { CallESAsCurrentUser, assetUsesObjects, getInstallationObject } from './index';
+import { assetUsesObjects, getInstallationObject } from './index';
 import { getObjects } from './get_objects';
 
 export async function installPackage(options: {
   savedObjectsClient: SavedObjectsClientContract;
   pkgkey: string;
-  asset: AssetType;
-  callCluster: CallESAsCurrentUser;
 }) {
-  const { savedObjectsClient, pkgkey, asset, callCluster } = options;
-  // install any assets (in ES, as Saved Objects, etc) as required. Get references to them
-  const toSave = await installAssets({
-    savedObjectsClient,
-    pkgkey,
-    asset,
-    callCluster,
-  });
-
-  if (toSave.length) {
-    // saved those references in the package manager's state object
-    const saved = await saveInstallationReferences({
+  const { savedObjectsClient, pkgkey } = options;
+
+  // Only install Kibana assets during package installation. All other asset types need special handling
+  const assetTypes: AssetType[] = [AssetType.visualization, AssetType.dashboard, AssetType.search];
+
+  let result = {};
+
+  // this is a for loop because using forEach with an async callback wouldn't wait for the
+  // awaits to resolve, and we'd return an empty result
+  for (let i = 0; i < assetTypes.length; i++) {
```

Comment:
Can also write this as

```suggestion
  for (const assetType of assetTypes) {
```